azurerm/internal/services/web/resource_arm_function_app.go

```diff
@@ -746,6 +788,51 @@ func expandFunctionAppSiteConfig(d *schema.ResourceData) web.SiteConfig {
 		siteConfig.HTTP20Enabled = utils.Bool(v.(bool))
 	}
 
+	if v, ok := config["ip_restriction"]; ok {
+		ipSecurityRestrictions := v.([]interface{})
+		restrictions := make([]web.IPSecurityRestriction, 0)
+		for i, ipSecurityRestriction := range ipSecurityRestrictions {
+			restriction := ipSecurityRestriction.(map[string]interface{})
+
+			ipAddress := restriction["ip_address"].(string)
+			vNetSubnetID := restriction["virtual_network_subnet_id"].(string)
+			if vNetSubnetID != "" && ipAddress != "" {
+				return siteConfig, fmt.Errorf(fmt.Sprintf("only one of `ip_address` or `virtual_network_subnet_id` can set set for `site_config.0.ip_restriction.%d`", i))
+			}
+
+			if vNetSubnetID == "" && ipAddress == "" {
+				return siteConfig, fmt.Errorf(fmt.Sprintf("one of `ip_address` or `virtual_network_subnet_id` must be set set for `site_config.0.ip_restriction.%d`", i))
+			}
+
+			ipSecurityRestriction := web.IPSecurityRestriction{}
+			if ipAddress != "" {
+				mask := restriction["subnet_mask"].(string)
+				if mask == "" {
+					mask = "255.255.255.255"
+				}
+				// the 2018-02-01 API expects a blank subnet mask and an IP address in CIDR format: a.b.c.d/x
+				// so translate the IP and mask if necessary
```

Comment:
Can we make this block behave like the API? ie remove the subnet mask property and expect the IP address to be in the CIDR format?

Reply:
Fixed